### ISSUE TITLE
build does not pickup changes in plugin script file without clean

### ISSUE THREAD

**Issue by ottobackwards** [zeek/zeek#1848]
When you run `make && sudo make install` in the build directory of a plugin generated with init-plugin, using the zeek cmake files and generated make files changes to script files are not picked up and installed.

### Steps to reproduce:
- Follow the QuickStart steps for ROT13 from the [documentation](https://docs.zeek.org/en/master/devel/plugins.html)
- after building and installing the plugin the first time `cd build && make && sudo make install`, read the __load__.zeek file and verify the contents ( `vi /usr/local/zeek/lib/zeek/plugins/Demo_Rot13/scripts/__load__.zeek`)
- in the main source directory, open `scripts/__load__.zeek` and edit, adding a comment `# this is a new comment`
- once again run  `cd build && make && sudo make install`
-  `vi /usr/local/zeek/lib/zeek/plugins/Demo_Rot13/scripts/__load__.zeek`
-  note that the change is not there ( even though it is in the linked script directory under build )
-  now run `cd build && make clean && make && make install`
-  `vi /usr/local/zeek/lib/zeek/plugins/Demo_Rot13/scripts/__load__.zeek`
-  note that the comment has now been added and deployed

### expectation
- changes to scripts should be picked up when running `make && make install` without requiring a clean and complete rebuild

### version
Reproduced with latest master, found in 4.1.1 originally for our plugins

**Comment by ottobackwards** [comment]
The issue is in `ZeekPluginDynamic.cmake`
The add_custom_command for _dist_output ( that calls zeek-plugin-create-package.sh ) only depends on the _plugin_lib.
If I change it to _also_ depend on the specific script that I'm modifying then the changes get picked up

```cmake
    add_custom_command(OUTPUT ${_dist_output}
            COMMAND ${BRO_PLUGIN_BRO_CMAKE}/zeek-plugin-create-package.sh ${_plugin_name_canon} ${_plugin_dist}
            WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
            DEPENDS ${_plugin_lib} "${BRO_PLUGIN_SCRIPTS}/__load__.zeek"
            COMMENT "Building binary plugin package: ${_dist_tarball_name}")
```

So I think the fix is to have it depend on each .zeek file in BRO_PLUGIN_SCRIPTS

@timwoj ?

I would think, that having zeek_plugin_script() to build up the scripts to be added would be consistent, since if you search for the files, it will only work at configure time anyways, and it would make it explicit that you would need to add a script to track a script.
It would have to be made to behave as it does now ( work but not depend ) if zeek_plugin_script wasn't found.

**Comment by ottobackwards** [comment]
I've verified that the zeek_plugin_script() approach works etc, I'll just go ahead and throw up the pr(s)

**Comment by ckreibich** [comment]
Resolved by zeek/cmake#33, closing.
